packages/langium/test/utils/cst-utils.test.ts

```diff
@@ -0,0 +1,28 @@
+/******************************************************************************
+ * Copyright 2022 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+import { createLangiumGrammarServices, EmptyFileSystem, Grammar } from '../../src';
+import { parseHelper } from '../../src/test';
+import * as c from '../../src/utils/cst-util';
+
+const services = createLangiumGrammarServices(EmptyFileSystem);
+const parser = parseHelper<Grammar>(services.grammar);
+
+describe('CST Utils', () => {
+
+    test('Find Leaf Node at Offset', async () => {
```

Comment:
Suggestion: It might be interesting to test edge case behavior in here as well. Your test covers the case where the cursor is at `<|>AB;`, but `A<|>B` and `AB<|>;` (which should return the leaf node for `;`) are also very important.

Reply:
added add'l test cases